submissions/SadegAshour/assignment8.js

```diff
@@ -0,0 +1,49 @@
+// Task 1
+const editor = {
+  title: 'my first blog',
+  getUpperTitle() {
+    return this.title.toUpperCase();
+  }
+};
+
+const getTitle = editor.getUpperTitle.bind(editor); // Bind the getUpperTitle method to the editor object
+
+console.log(getTitle()); // This will log "MY FIRST BLOG"
+
+// Write code that ensures getTitle() works as expected and logs "MY FIRST BLOG"
+//-------------------------------------------------
+// Task 2
+const formHandler = {
+  value: "initial",
+  onChange(newValue) {
+    this.value = newValue;
+  },
+};
+
+function simulateInputChange(callback) {
+  callback("updated");
+}
+
+simulateInputChange(formHandler.onChange); // initial cause this lost binding so the value is not updated
+// ❓ Now log formHandler.value
+console.log(formHandler.value); // "initial"
+
+// Fix the bug using either .bind() or an arrow function (your choice).
+simulateInputChange(formHandler.onChange.bind(formHandler));
```

Comment:
Please use arrow functions as another solution without editing the original code